Spokestack/Spokestack.swift

```diff
@@ -0,0 +1,134 @@
+//
+//  Spokestack.swift
+//  Spokestack
+//
+//  Created by Noel Weichbrodt on 9/23/20.
+//  Copyright © 2020 Spokestack, Inc. All rights reserved.
+//
+
+import Foundation
+
+/// This class combines all Spokestack modules into a single component to provide a unified interface to the library's ASR, NLU, and TTS features. Like the individual modules, it is configurable using a fluent builder pattern, but it provides a default configuration; only a few parameters are required from the calling application, and those only for specific features noted in the documentation for the builder's methods.
+///
+/// The default configuration of this class assumes that the client application wants to use all of Spokestack's features, regardless of their implied dependencies or required configuration. If a prerequisite is missing at build time, the individual module may throw an error when called.
+///
+/// This class will run in the context of the caller. The subsystems themselves may use the configured dispatch queues where appropriate to perform intensive tasks.
+///
+/// - SeeAlso:
+/// `SpeechPipeline`, `NLUTensorflow`, `TextToSpeech`
+///
+@objc public class Spokestack: NSObject {
+    /// This is the client entry point to the Spokestack voice input system.
+    @objc public var pipeline: SpeechPipeline?
+    /// This is the client entry point for the Spokestack Text to Speech (TTS) system.
+    @objc public var tts: TextToSpeech?
+    /// This is the client entry point for the Spokestack BERT NLU implementation.
+    @objc public var nlu: NLUTensorflow?
+    /// Maintains global state for the speech pipeline.
+    @objc public var context: SpeechContext?
+    /// Configuration properties for Spokestack modules.
+    @objc public var configuration: SpeechConfiguration
+    
+    private var delegates: [SpokestackDelegate]
+    
+    /// This constructor is intended for use only by the `SpokestackBuilder`.
+    /// - Parameters:
+    ///   - delegates: Delegate implementations of `SpokestackDelegate` that receive Spokesatck module events.
```

Comment:
I think I picked up "module" from the original pipeline spec. I do kind of prefer it to "subsystem" just because it is shorter and reinforces the modular design ethic that we've tried for.